lib/ecto/uuid.ex

```diff
@@ -29,9 +29,9 @@ defmodule Ecto.UUID do
   catch
     :error -> :error
   else
-    casted ->
-      {:ok, casted}
+    casted -> {:ok, casted}
   end
+  def cast(<< _::128 >> = binary), do: binary |> encode() |> cast()
```

Comment:
It should be safe to do `{:ok, encode(binary)}` here - we know encode returns the proper format.